crates/prover/src/client/proving.rs

```diff
@@ -67,53 +69,152 @@ where
         oracle_client,
         hint_client,
     ));
-    // todo: run hok witgen if hokking
-
     // Instantiate oracles
     let blob_provider = OracleBlobProvider::new(preimage_oracle.clone());
     // Run witness generation with oracles
-    let (proof_journal, mut witness, da_witness) = match l1_node_address.is_some() {
-        false => {
-            witgen::run_witgen_client(
-                preimage_oracle.clone(),
-                10 * 1024 * 1024, // default to 10MB chunks
-                blob_provider,
-                EthereumDataSourceProvider,
-                proving.payout_recipient_address.unwrap_or_default(),
-                precondition_validation_data_hash,
-                execution_cache.clone(),
-                stitched_boot_info.clone(),
-            )
-            .await
-            .context("Failed to run vec witgen client.")
-            .map_err(ProvingError::OtherError)
-            .map(|(_, j, w)| (j, w, None))?
-        }
-        true => {
-            witgen::run_hokulea_witgen_client(
-                preimage_oracle.clone(),
-                10 * 1024 * 1024, // default to 10MB chunks
-                blob_provider,
-                proving.payout_recipient_address.unwrap_or_default(),
-                precondition_validation_data_hash,
-                execution_cache.clone(),
-                stitched_boot_info.clone(),
-            )
-            .await
-            .context("Failed to run vec witgen client.")
-            .map_err(ProvingError::OtherError)
-            .map(|(j, w, e)| (j, w, Some(e)))?
-        }
-    };
+    let (proof_journal, witness, extra_frames, extra_proofs) =
+        match (proving.use_hokulea(), proving.use_hana()) {
+            (false, false) => {
+                witgen::run_witgen_client(
+                    preimage_oracle.clone(),
+                    10 * 1024 * 1024, // default to 10MB chunks
+                    blob_provider,
+                    EthereumDataSourceProvider,
+                    proving.payout_recipient_address.unwrap_or_default(),
+                    precondition_validation_data_hash,
+                    execution_cache.clone(),
+                    stitched_boot_info.clone(),
+                )
+                .await
+                .context("Failed to run kona vec witgen client.")
+                .map_err(ProvingError::OtherError)
+                .map(|(_, j, w)| (j, w, vec![], vec![]))?
+            }
+            (true, _) => {
```

Comment:
I think you can use `clap`'s `conflict_with to make somehow make sure only hokulea or eigen is enabled, so you can't accidentally activate both and then have to default to one.